hcl/hcl.go

```diff
@@ -1002,6 +1023,115 @@ func parseStack(evalctx *eval.Context, stack *Stack, stackblock *ast.Block) erro
 	return errs.AsError()
 }
 
+func checkHasSubBlocks(block *ast.Block, blocks ...string) error {
```

Comment:
The function is called `checkHasSubBlocks()` but it also fails if the block has attributes and fails if it has labels. Maybe you can make it only checking for sub-blocks and create another function that does the complete check for this kind of block?

Reply:
hmm yeah it does way too much things then naming it was complicated, will improve.